support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -316,10 +303,51 @@ function ChangeButton({ geoId }: ChangeButtonProps) {
 type Props = {
 	geoId: GeoId;
 	appConfig: AppConfig;
-	useStripeExpressCheckout?: boolean;
 };
 export function Checkout({ geoId, appConfig }: Props) {
 	const { currencyKey } = getGeoIdConfig(geoId);
+	const searchParams = new URLSearchParams(window.location.search);
+
+	/** Get and validate product */
+	const productParam = searchParams.get('product');
+	const productKey =
+		productParam && isProductKey(productParam) ? productParam : undefined;
+	const product = productKey && productCatalog[productKey];
+	if (!product) {
+		return <div>Product not found</div>;
+	}
+
+	/**
+	 * Get and validate ratePlan
+	 * TODO: This type should be more specific e.g. `ProductRatePlanKey<P>`.
+	 * Annoyingly the TypeScript for this is a little fiddly due to the
+	 * API being completely based on literals, so we've left it as `string`
+	 * although we do validate it is a valid ratePlan for this product
+	 */
+	const ratePlanParam = searchParams.get('ratePlan');
+	const ratePlanKey =
+		ratePlanParam && ratePlanParam in product.ratePlans
+			? ratePlanParam
+			: undefined;
+	const ratePlan = ratePlanKey && product.ratePlans[ratePlanKey];
+	if (!ratePlan) {
+		return <div>Rate plan not found</div>;
+	}
+
+	/** Get and validate price */
+	let price: number | undefined;
+	if (productKey === 'Contribution') {
+		const priceParam = searchParams.get('price');
+		price = priceParam ? parseInt(priceParam, 10) : undefined;
```

Comment:
Does 10 need to b specified inside parseInt?

Reply:
[From the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#description)

> Be careful — this does not always default to 10!